joplin/base/models.py

```diff
@@ -482,6 +481,52 @@ class DepartmentPageDirector(Orderable):
         FieldPanel('about'),
     ]
 
+class OfficialDocumentPage(JanisPage):
+    janis_url_page_type = "official_document"
+    base_form_class = custom_forms.OfficialDocumentPageForm
+
+    description = models.TextField(blank=True)
+
+    department = models.ForeignKey(
+        'base.DepartmentPage',
+        on_delete=models.PROTECT,
+        verbose_name='Select a Department',
+        blank=True,
+        null=True,
+    )
+
+    content_panels = [
+        FieldPanel('title_en'),
+        FieldPanel('title_es'),
+        FieldPanel('title_ar'),
+        FieldPanel('title_vi'),
+        FieldPanel('description'),
+        InlinePanel('topics', label='Topics'),
+        InlinePanel('related_departments', label='Related Departments'),
+        InlinePanel('official_documents', label="Documents", heading="Entries will be listed by document date (newest first)."),
+    ]
+
+class OfficialDocumentPageOfficialDocument(Orderable):
```

Comment:
This naming is confusing to me. Is this supposed to be referring to a particular document that you'd include on a 'list of documents' page?

A clarifying docstring would be great for when the code gets revisited later. Realizing this follows some existing naming conventions in the current codebase...

Reply:
I think current naming conventions are more along the lines of:
`ServicePage` - a page about a service
`Contact`- the contact info for something
`ServicePageContact` - The relationship between the ServicePage and the Contact

It seems as though this naming is more like
`OfficialDocumentPage` - the official document page
`OfficialDocumentPageOfficialDocument` - the documents that go on that page

Doesn't quite fit the (completely undocumented and not actually existing) naming conventions we have, but I see no issue with keeping it as is.

Reply:
Ah ok I see the naming pattern now.

I was going off this kind of pattern: `DepartmentPageDirector` is a Director for a Department Page. If many pages had Directors, then you'd want a standalone `Director` model and then a relationship which would be `DepartmentPageDirector`. But only `OfficialDocumentPages` will have `OfficialDocuments`. In case the word `Document` is later used to refer to built-in document uploading, I went with the more verbose `OfficialDocument`.

I'll add documentation.